src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -8498,7 +8487,7 @@ protected override void VisitForEachExpression(BoundForEachStatement node)
 
             MethodSymbol? reinferredGetEnumeratorMethod = null;
 
-            if (node.EnumeratorInfoOpt?.GetEnumeratorMethod is { IsExtensionMethod: true, Parameters: var parameters } enumeratorMethod)
+            if (node.EnumeratorInfoOpt?.GetEnumeratorInfo.Method is { IsExtensionMethod: true, Parameters: var parameters } enumeratorMethod)
             {
                 // this is case 7
                 var (method, results, _) = VisitArguments(
```

Comment:
Note to self: On monday, fix the bug 2 lines below and add a test for nullable with params/default parameter extension methods.